doc/user/sideeffect.xml

```diff
@@ -46,79 +46,157 @@
 
  <para>
 
- If &SCons; is unaware that a build step produces an extra file,
- the &SideEffect; method can be used to identify it,
- so that the file can be used as a dependency in subsequent build steps.
- However, the primary use for the &SideEffect; method
+ Sometimes the way an action is defined causes effects on files
+ that &SCons; does not recognize as targets. The &f-link-SideEffect;
+ method can be used to informs &SCons; about such files.
+ This can be used just to flag a dependency for use in subsequent
+ build steps, although there is usually a better way to do that.
+ The primary use for the &SideEffect; method
  is to prevent two build steps from simultaneously modifying the same file.
```

Comment:
This can actually be just opening, reading, or writing, especially on windows that would cause build errors.

Reply:
Also perhaps an example could be a .lock file (some tools use these to keep from running 2x processes at the same time)

Reply:
A lock file is an example of a sideeffect

Reply:
`two build steps from simultaneously modifying the same file` to modifying or accessing in a way would impact another tool trying to access the file.. For win32. just opening it can be enough to block/break another process trying to access it.

Reply:
How about 1/2? :)  I'm not feeling a natural way to mention a lock file here.